server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -649,7 +631,67 @@ private static SnapshotsInRepo sortSnapshots(
         return new SnapshotsInRepo(resultSet, snapshotInfos.size(), allSnapshots.size() - resultSet.size());
     }
 
-    private static List<SnapshotInfo> filterBySLMPolicies(List<SnapshotInfo> snapshotInfos, String[] slmPolicies) {
+    private static Predicate<SnapshotInfo> buildAfterPredicate(
+        GetSnapshotsRequest.SortBy sortBy,
+        String after,
+        SortOrder order,
+        @Nullable String snapshotName,
+        @Nullable String repoName
+    ) {
+        final Predicate<SnapshotInfo> isAfter;
+        switch (sortBy) {
+            case START_TIME:
+                isAfter = filterByLongOffset(SnapshotInfo::startTime, Long.parseLong(after), snapshotName, repoName, order);
```

Comment:
I think we should add explicit error handling when parsing the arguments, perhaps to the request validation or here. That would allow the message to indicate that it is the `after_value` causing the problem.

Also, I think `< 0` is illegal so we could guard against that.

Similarly for duration, indices and failed shards.

Reply:
I just realized we have the safe problem for the `after` value as well (though due to the way we encode things there it's less likely). May I push this into a follow-up? :) Then I can just deal with both in one go and make it a little nicer without blowing this one up.

Reply:
That said, I refactored this a little to build the predicate early on now. This makes sure that we at least don't run any actual requests unless we have a valid predicate. Refactoring this nicer and working in a fix for the `after` param + easy-to-read validation I'd still like to push to a follow-up though if possible :)

Reply:
I think that for `after` we expect it to be handle opaquely by clients so if they mess with it, I am OK to have a worse error (though I will not object to improving it).

I am OK with pushing the error handling to a separate PR.